torch/nn/utils/parametrize.py

```diff
@@ -0,0 +1,361 @@
+        # the next time we call module[name]
+        self()
+
+    def remove(self, leave_parametrized=False):
+        r"""Removes parameterization. If it is the last parametrization,
+        the parametrized parameter named ``name+'_orig'`` is deleted and
+        the buffer ``name`` is turned into a parameter.
+
+        Args:
+            leave_parametrized (Bool, optional): If it is the last parametrization,
+            leave the parametrized tensor or the original (unparametrized) tensor
+            after deletion. It shall only be True whenever the parametrization does
+            not change the size of the tensor.
+            Default: False
+
+        Note:
+            If one leaves the unparametrized tensor, the shape of the parameter ``name``
+            might change
+        """
+        if not hasattr(self, "_tensor_name"):
+            raise ValueError("Parametrization has to be applied\
+                              before it can be removed")
+
+        param = get_parametrization(self._module, self._tensor_name)
+
+        if len(param) == 0:
+            raise ValueError("No parametrizations found on {}".format(self._tensor_name))
+        elif leave_parametrized and len(param) != 1:
+            raise ValueError("leave_parametrized=True only takes effect whenever"
+                             "there are exactly one parametrizations. {} parametrizations"
+                             "where found on {}".format(len(param), self._tensor_name))
+
+        # If it is the last one, we delete the created tensors
+        if len(param) == 1:
+            # Lezcano. test_pruning_id_consistency asserts that putting a constraint
+            # and removing it leaves the id of the parameter invariant
+            # This can only be achieved if the parametrization does not change the
+            # dimensions of the tensor. Is this invariance of id really necessary?
+            orig = self._module._parameters[self._tensor_name + "_orig"]
+            if leave_parametrized:
+                if self.mode == "forward":
+                    # We update the parametrization first to return it updated
+                    setattr(self._module, self._tensor_name, self.apply_param())
+                t = getattr(self._module, self._tensor_name)
+                if t.size() != orig.size():
+                    raise ValueError("We only support to leave_parametrized=True when "
+                                     "the parametrization does not change"
+                                     "the dimensions of the tensor.")
+                orig.data = t
+
+            del self._module._buffers[self._tensor_name]
+            del self._module._parameters[self._tensor_name + "_orig"]
+            self._module.register_parameter(self._tensor_name, orig)
+
+        # Remove hook
+        self._hook_handle.remove()
+
+
+def update_auto_parametrization(module, name, parametrizations):
```

Comment:
Doesn't a module already store its parametrizations as hooks without the need to pass them in separately as args?

Reply:
Yep! Corrected now.